src/models/file.ts

```diff
@@ -148,8 +148,12 @@ export class TargetFile {
     this.unrecognizedFields = opts.unrecognizedFields || {};
   }
 
-  get custom(): JSONValue {
-    return this.unrecognizedFields['custom'];
+  get custom(): Record<string, unknown> {
+    const custom = this.unrecognizedFields['custom'];
+    if (!custom || Array.isArray(custom) || !(typeof custom === 'object')) {
```

Comment:
nit: It might not be immediately obvious to someone with less js experience that `!(typeof custom === 'object')` only does the expected thing because you check `Array.isArray(custom) ` first. 

This is a little ugly but could check that the object is really an object and not an array object and get rid of the array check? 

```suggestion
    if (!custom || Object.prototype.toString.call(custom) !== '[object Object]') {
```

Reply:
@feelepxyz unfortunately the TypeScript compiler doesn't understand that check, so you'd still need to do some more filtering to get the type properly narrowed:

<img width="852" alt="image" src="https://user-images.githubusercontent.com/398027/214596889-41058a7f-ace4-4c3a-9bad-2cc9b976c923.png">

I agree your suggestion is a bit more intuitive to the reader, but this is one of those situations where we need to cater to the compiler (which does understand `isArray` and `typeof`)

Reply:
ahh gotcha! def 👍 on leaning on the type system here